src/scripts/main.js

```diff
@@ -1,3 +1,1004 @@
 'use strict';
 
-// write your code here
+const page = document.querySelector('.page');
+const switcherStart = document.querySelector('.start');
+const switcherScore = document.querySelector('.game-score');
+
+let workingArray = Array(16).fill(null);
+let changeIndicator = true;
+let scoreCounter = 0;
+let disableFunctions = false;
+
+function generateInitialValues() {
+  let positionX;
+  let positionY;
+  const startValueX = Math.floor(Math.random() * 9);
+  const startValueY = Math.floor(Math.random() * 9);
+
+  do {
+    positionX = Math.floor(Math.random() * 16);
+    positionY = Math.floor(Math.random() * 16);
+  } while (positionX === positionY);
+
+  if (startValueX !== 0) {
+    page.querySelectorAll('.field-cell')[positionX]
+      .classList.add('field-cell--2');
+    workingArray[positionX] = 2;
+  } else {
+    page.querySelectorAll('.field-cell')[positionX]
+      .classList.add('field-cell--4');
+    workingArray[positionX] = 4;
+  }
+
+  if (startValueY !== 0) {
+    page.querySelectorAll('.field-cell')[positionY]
+      .classList.add('field-cell--2');
+    workingArray[positionY] = 2;
+  } else {
+    page.querySelectorAll('.field-cell')[positionY]
+      .classList.add('field-cell--4');
+    workingArray[positionY] = 4;
+  };
+
+  switcherStart.classList.remove('start');
+  switcherStart.classList.add('restart');
+  switcherStart.textContent = 'Restart';
+  page.querySelector('.message-start').classList.add('hidden');
+}
+
+function generateTwoOrFour() {
+  let findAllIndices = [];
+
+  findAllIndices = [];
+
+  for (let i = 0; i < workingArray.length; i++) {
+    if (workingArray[i] !== null) {
```

Comment:
pls [see falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
    if (workingArray[i]) {
```